### ISSUE TITLE
free(): double free detected in tcache 2

### ISSUE THREAD

**Issue by mmmmimic** [HuXiaoling/TopoLoss#19]
Hi Xiaoling! 

Thank you for the impressive work! I had some trouble when incorporating the TopoLoss into my model training. Specifically, I have two questions
- Is the function getTopoLoss() intended for processing single image pairs? I would kindly like to know the shape of the two inputs, likelihood_tensor and gt_tensor, in PyTorch. 
- I met an error called "free(): double free detected in tcache 2" during training. Do you happen to have any idea about this error?

Thank you in advance!

**Comment by HuXiaoling** [comment]
@mmmmimic, thanks for your interest in our work! What do you mean by 'processing single image pairs'? The shapes of the two inputs are dependent on your data -- once the patch contains meaningful topological information.

Sorry that I haven't never seen such errors before.

**Comment by mmmmimic** [comment]
Thank you! Could the getTopoLoss() handle inputs in batch? If I am computing the TopoLoss for a multi-class segmentation task, would there be an efficient way to compute the loss for images in a batch?

**Comment by HuXiaoling** [comment]
The function getTopoLoss() was originally designed for single image pairs.

**Comment by mmmmimic** [comment]
Thank you!
